### DISCUSSION TITLE
RuntimeError: Unsupported fstring syntax happens with xonsh 0.18.3 and Python 3.12

### DISCUSSION THREAD

**Discussion by halloleo** [xonsh/xonsh#5708] [General]
When I do in xonsh 0.18.3

    $VAR = "Hallo"
    txt = f"$(VAR} world!"

I get the error "_RuntimeError: Unsupported fstring syntax_". 

I saw https://github.com/xonsh/xonsh/discussions/5227 which reports this issue for xonsh 0.14. And then there is https://github.com/xonsh/xonsh/issues/5166. 

So it is a Python 3.12, not a Xonsh issue?  Upgrading to 3.13 doesn't seem to fix it though.

For the moment IMGO the best way to work around this is [sbliven's suggestion][1]:

    txt = f"{__xonsh__.env['VAR']} world!"

  [1]: https://github.com/xonsh/xonsh/discussions/5227#discussioncomment-8000765

**Comment by anki-code** [comment]
It's xonsh issue #5166

**Reply by halloleo** [reply]
Thanks @anki-code. 

The way I read the issue, when looking over it, was that it is caused by a change in Python. But of course we have to adapt Xonsh to this. %-) 

I imaging there is no easy fix!
